src/main/java/ledger/user_interface/ui_controllers/MainPageController.java

```diff
@@ -112,5 +334,201 @@ public void initialize(URL fxmlFileLocation, ResourceBundle resources) {
+            System.out.println("Error of loading all transactions into list view: " + e.getMessage());
+        }
+    }
+
+    private void handleDeleteTransactionFromTableView() {
+        try {
+            int selectedIndex = this.transactionTableView.getSelectionModel().getSelectedIndex();
+            if (selectedIndex >= 0) {
+                TransactionModel model = (TransactionModel) this.transactionTableView.getItems().get(selectedIndex);
+                Transaction transactionToDelete = model.getTransaction();
+
+                TaskWithArgs<Transaction> task = DbController.INSTANCE.deleteTransaction(transactionToDelete);
+                task.startTask();
+                task.waitForComplete();
+            } else {
+                System.out.println("No transactions deleted. Selected index: " + selectedIndex);
+            }
+        } catch (StorageException e) {
+            System.out.println("Error deleting transaction: " + e.getMessage());
+        }
+    }
+
+    public class TransactionModel {
+        private int id;
+        private String amount;
+        private String date;
+        private String payeeName;
+        private String typeName;
+        private String tagNames;
+        private String pending;
+        private Transaction transaction;
+
+        public TransactionModel(Transaction transaction) {
+            this.id = transaction.getId();
+            String amountInCents = String.valueOf(transaction.getAmount());
+            String dollars = amountInCents.substring(0, amountInCents.length() - 2);
+            String cents = amountInCents.substring(amountInCents.length() - 2, amountInCents.length());
+            this.amount = "$" + dollars + "." + cents;
+            if (transaction.getDate() != null) {
+                this.date = transaction.getDate().toString();
+            } else {
+                this.date = "";
+            }
+            if (transaction.getPayee() != null) {
+                this.payeeName = transaction.getPayee().getName();
+            } else {
+                this.payeeName = "";
+            }
+            if (transaction.getType() != null) {
+                this.typeName = transaction.getType().getName();
+            } else {
+                this.typeName = "";
+            }
+            this.tagNames = "";
+            if (transaction.getTagList() != null) {
+                List<Tag> tags = transaction.getTagList();
+                for (int i = 0; i < tags.size(); i++) {
+                    this.tagNames += tags.get(i).getName();
+                    if (i != tags.size() - 1) {
+                        this.tagNames += ", ";
+                    }
+                }
+            }
+            if (transaction.isPending()) {
+                this.pending = "Pending";
+            } else {
+                this.pending = "Cleared";
+            }
+
+            this.transaction = transaction;
+        }
+
+        public int getId() {
```

Comment:
Tried that. With the way the table view generates its data, it's not possible to do the necessary data manipulation to make it presentable to the user.